src/custom/state/swap/TradeGp.ts

```diff
@@ -0,0 +1,151 @@
+import { CanonicalMarketParams, getCanonicalMarket } from '@src/custom/utils/misc'
```

Comment:
the majority of the logc (unchanged logically) that WAS in `trade/extensions.ts` is now here as a class with the necessary methods coded in. This solves the need for the previous uni trade to be mixed in to meet type requirements.